src/vcpkg/commands.export.cpp

```diff
@@ -389,31 +389,60 @@ namespace vcpkg::Export
         ret.prefab_options.enable_maven = Util::Sets::contains(options.switches, OPTION_PREFAB_ENABLE_MAVEN);
         ret.prefab_options.enable_debug = Util::Sets::contains(options.switches, OPTION_PREFAB_ENABLE_DEBUG);
         ret.maybe_output = Util::lookup_value_copy(options.settings, OPTION_OUTPUT);
-        ret.output_dir = Util::lookup_value(options.settings, OPTION_OUTPUT_DIR)
-                             .map([&](const Path& p) { return paths.original_cwd / p; })
-                             .value_or(paths.root);
         ret.all_installed = Util::Sets::contains(options.switches, OPTION_ALL_INSTALLED);
 
-        if (ret.all_installed)
+        if (paths.manifest_mode_enabled())
         {
+            auto output_dir_opt = Util::lookup_value(options.settings, OPTION_OUTPUT_DIR);
+
+            // --output-dir is required in manifest mode
+            if (!output_dir_opt.has_value())
+            {
+                msg::println_error(msgCmdExportMissingArgument, msg::option = "--output-dir");
```

Comment:
Can we reuse `msgMissingOption` instead?

```suggestion
                msg::println_error(msgMissingOption, msg::option = "output-dir");
```